### DISCUSSION TITLE
Top speed historic chart?

### DISCUSSION THREAD

**Discussion by detritu5** [teslamate-org/teslamate#1237] [Ideas]
Is there any option to see the top speed ever registered in the database? Or can I add a Historical chart to see the best top speed with the date registered? It would be interested

**Comment by ronajon** [comment]
i second that. 
a grafana page with all sorts of top<information> would be great

top speed
top distance per day
top charging speeds
top ...

**Comment by dennisbos84** [comment]
For top-speed 'positions' you can try on the cli: 
`docker exec -it teslamate_database_1 psql -U teslamate -c "select date, latitude, longitude, speed from positions where speed is not null order by speed desc limit 10;"`

Or for the max-speed per 'drive':
`docker exec -it teslamate_database_1 psql -U teslamate -c "select date(a.start_date) as date, date_trunc('second', a.start_date::time) as start_time, date_trunc('second', a.end_date::time) as end_time, a.speed_max, a.power_max, b.name as start_location, c.name as end_location from drives a left join addresses b on a.start_address_id = b.id left join addresses c on a.end_address_id = c.id where a.speed_max is not null order by a.speed_max desc limit 10;"`

**Comment by DrMichael** [comment]
Well, one could do something like
![image](https://user-images.githubusercontent.com/8087682/105187291-e85f8a80-5b32-11eb-9301-46dfb672e64f.png)

![image](https://user-images.githubusercontent.com/8087682/105187353-fc0af100-5b32-11eb-82bd-0b5cd5b00690.png)

**Comment by detritu5** [comment]
> Well, one could do something like
> ![image](https://user-images.githubusercontent.com/8087682/105187291-e85f8a80-5b32-11eb-9301-46dfb672e64f.png)
> 
> ![image](https://user-images.githubusercontent.com/8087682/105187353-fc0af100-5b32-11eb-82bd-0b5cd5b00690.png)

Yeah that's great!! could you share the json code?

**Comment by DrMichael** [comment]
Sure...Comments welcome.

[top-values-by-period.zip](https://github.com/adriankumpf/teslamate/files/5843116/top-values-by-period.zip)

**Comment by rsunny2012** [comment]
what is the location of the file to edit on my docker ubuntu image ?

**Comment by M1kep** [comment]
Not really sure what's changed, but I've updated one of the date_trunc calls that was failing for me
[top-values-by-period-fixed.json.zip](https://github.com/adriankumpf/teslamate/files/9294605/top-values-by-period-fixed.json.zip)
